jumanji/environments/logic/sudoku/generator.py

```diff
@@ -0,0 +1,118 @@
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Copyright 2022 InstaDeep Ltd. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.i
+
+import abc
+
+import chex
+import jax
+import jax.numpy as jnp
+
+from jumanji.environments.logic.sudoku.types import State
+from jumanji.environments.logic.sudoku.utils import get_action_mask
+
+
+class Generator(abc.ABC):
+    @abc.abstractmethod
+    def __call__(self, key: chex.PRNGKey) -> State:
+        """Call method responsible for generating a new state.
+
+        Args:
+            key: jax random key for any stochasticity used in the instance
+                generation process.
+
+        Returns:
+            A Sudoku State.
+        """
+        pass
+
+
+class DummyGenerator(Generator):
+    """Generates always the same board, for debugging purpose."""
+
+    def __init__(
+        self,
+    ) -> None:
+        board = jnp.array(
+            [
+                [0, 0, 0, 8, 0, 1, 0, 0, 0],
```

Comment:
Are these the same ones used in test files/init file? In which case maybe add them to constants?

Reply:
I agree, added them to the constants!